### PR TITLE
Add new module digital_ocean_sshkey_facts

### PR DESCRIPTION
**Pull request by pmarques**
##### SUMMARY

Add module to fetch Digital Ocean SSH Keys facts.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

 - cloud/digital_ocean/digital_ocean_sshkey_facts

### CODE REVIEW

**Comment by pmarques** [comment]
The PR with **do_sshkeys** module is https://github.com/ansible/ansible/pull/18930

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py
Comment (gundalow) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py]: Might be useful to show how these facts should be used, e.g. a more real world example

Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py]: This module was born as part of the do_sshkeys and end up using this to check keys in a cluster. So, for my use case it is just

```
- debug: var= result
```

The **RETURN** explains the data structure of *result* and so, this can be used as any other variable to test / conditional run tasks if someone needs.

**Comment by pmarques** [comment]
I didn't got the problem on shippable o0

**Comment by BondAnthony** [comment]
@pmarques would you be open to refactoring your facts modules to use `../../../module_utils/digital_ocean.py` in an effort to reduce duplication of code?

**Comment by pmarques** [comment]
@BondAnthony sure, (creating) modules_utils was was part of the initial plan.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py]: Could you add the oauth_token parameter to the example

Comment (BondAnthony) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py]: Also could you add to the example... I think something like this would be valuable for users to see. Hopefully it will help explain when to use this module over the `digital_ocean_sshkey` module.

```
    - digital_ocean_sshkey_facts:
         oauth_token: "{{ my_do_key }}"

    - set_fact:
        pubkey: "{{ item.public_key }}"
      with_items: "{{ ssh_keys|json_query(ssh_pubkey) }}"
      vars:
        ssh_pubkey: "[?name=='ansible_crtl']"

    - debug:
        msg: "{{ pubkey }}"
```

I'm not sure the `register` command provides much value since your returning ansible_facts. Users can directly access `ssh_keys` as an ansible fact but this module spits out an object that will need a filter applied to it. Still useful if your trying to get a pub-key for additional operations on your servers. Might be helpful when seeding more keys on already running droplets.

**Review by BondAnthony — CHANGES_REQUESTED** [review]
Refinement to the example to show better use case. Looks good other than that minor detail. Thank you for the work!

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py]: @BondAnthony Nice example, I changed it as you suggested

**Comment by BondAnthony** [comment]
Thank you @pmarques.

@gundalow does this example look better?

**Review by BondAnthony — APPROVED** [review]
shipit

**Comment by pmarques** [comment]
@BondAnthony after so many changes and tests I started a Digital Ocean integration tests, https://github.com/ansible/ansible/pull/29960

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (gundalow) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py:107]: When would this be needed?
I assume (maybe incorrectly) that we are always talking to `https://api.digitalocean.com` which *should* have a valid cert.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (gundalow) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py:108]: Please add this to the  `DOCUMENTATION` block.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (gundalow) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py:58]: FYI comments will not be rendered in the RETURN table, if this is for future developers of this module, then that's fine.

If however this link is for users of the module then you may wish to consider adding it to `DOCUMENTATION:` `notes:`

http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py:107]: True, As far as I remember that option was requested by someone in a PR that I not able to find :worried:.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py:58]: This is a comment/reference to check the API examples.

**Review by BondAnthony — APPROVED** [review]
LGTM

**Comment by thaumos** [comment]
can we get two "shipits" from @BondAnthony and @pmarques just for documenting this?

**Comment by BondAnthony** [comment]
shipit

**Review by gundalow — APPROVED** [review]

**Comment by pmarques** [comment]
@thaumos Can I mark this with "shipit"? I'm implemented this. Anyway waiting for a review from @ryansb

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
Comment (ryansb) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py]: s/crtl/ctrl/

**Review by ryansb — APPROVED** [review]
Looks good - minor docs misspelling that can be fixed in a follow-up PR.

shipit

**Comment by pmarques** [comment]
ups... removed the label 😞 @BondAnthony @ryansb  😞

**Comment by ryansb** [comment]
Shipped manually since that was such a trivial change

**Pull request merged**
